crates/oxc_semantic/src/builder.rs

```diff
@@ -1586,6 +1546,20 @@ impl<'a> SemanticBuilder<'a> {
             AstKind::ExportSpecifier(s) if s.export_kind.is_type() => {
                 self.current_reference_flag = ReferenceFlag::Type;
             }
+            AstKind::BindingIdentifier(_) => {
+                if let Some(node) = self.nodes.parent_node(self.current_node_id) {
+                    if let AstKind::Class(class) = node.kind() {
+                        if class.is_expression() {
+                            // We must to bind class expression when enter BindingIdentifier,
+                            // because we should add binding to current scope
+                            let previous_node_id = self.current_node_id;
+                            self.current_node_id = node.id();
+                            class.bind(self);
+                            self.current_node_id = previous_node_id;
+                        }
```

Comment:
Bind `ClassExpression` in `BindingIdentifier` to add the binding to the correct scope